### PR TITLE
fix: #586 #740

### PR DESCRIPTION
**Pull request by Jocs**
| Q                | A
| ---------------- | ---
| Bug fix?         | yes/no
| New feature?     | yes/no
| BC breaks?       | yes/no
| Deprecations?    | yes/no
| New tests added? | yes/not needed
| Fixed tickets    | #586  #740 
| License          | MIT

### Description

[Description of the bug or feature]

nested math block. and inline math parse error.

--

### CODE REVIEW

File: src/muya/lib/parser/rules.js [hunk 1]
```diff
@@ -10,9 +10,10 @@ export const beginRules = {
 }
 
 export const inlineRules = {
-  'backlash': /^(\\)([\\`*{}\[\]()#+\-.!_>~:\|\<\>]{1})/,
+  'backlash': /^(\\)([\\`*{}\[\]()#+\-.!_>~:\|\<\>$]{1})/,
```
Comment (fxha) [inline src/muya/lib/parser/rules.js]: Can you please separate original, patched and new rules like [here](https://github.com/marktext/marktext/blob/0a3e05c0c12a32a198a936d5e1c0901d160f01ae/src/muya/lib/parser/marked/inlineRules.js#L32)?

File: src/muya/lib/parser/rules.js [hunk 2]
```diff
@@ -5,12 +5,14 @@ export const beginRules = {
...
-  'reference_definition': /^( {0,3}\[)([^\]]+?)(\\*)(\]: *)(<?)([^\s>]+)(>?)(?:( +)(["'(]?)([^\n"'\(\)]+)\9)?( *)$/
+  'reference_definition': /^( {0,3}\[)([^\]]+?)(\\*)(\]: *)(<?)([^\s>]+)(>?)(?:( +)(["'(]?)([^\n"'\(\)]+)\9)?( *)$/,
+
+  // extra syntax (not belogs to GFM)
+  'multiple_math': /^(\$\$)$/
 }
 
 export const inlineRules = {
-  'backlash': /^(\\)([\\`*{}\[\]()#+\-.!_>~:\|\<\>]{1})/,
+  'backlash': /^(\\)([\\`*{}\[\]()#+\-.!_>~:\|\<\>$]{1})/,
```
Comment (fxha) [inline src/muya/lib/parser/rules.js]: Could you please add a comment here that the line is patched? I would it make easier for further modifications.

**Comment by Jocs** [comment]
@fxha updated this pr, pls review again.

**Comment by fxha** [comment]
This PR doesn't fix #740 because we cannot escape `$`. #586 is fixed :+1:

**Comment by Jocs** [comment]
屏幕快照 2019-03-25 下午8 20 04

I can escape `$`?

**Comment by fxha** [comment]
Yes, my bad because `\` followed by `$` result in `\$$`. Do you want to disable autocompletion if it leaded by `\`?

**Comment by Jocs** [comment]
>Yes, my bad because \ followed by $ result in \$$. Do you want to disable autocompletion if it leaded by \?

haha, it's not a big problem but it's a good improvement.

**Pull request merged**